docs/curve_dao/fee-collection-distribution/curve-burner/Hooker.md

```diff
@@ -0,0 +1,767 @@
+- **Cooldown Management**: Compensation strategies include cooldown periods to prevent abuse and ensure fair distribution of rewards. The `duty_counter` helps manage these cooldown periods by tracking the epochs in which compensations are made.
+- **Security and Access Control**: Certain functions, like `set_hooks` and `one_time_hooks`, are restricted to the contract owner to maintain security and control over the contract's behavior.
+
+Hooks need to be added to the contract via the [`set_hooks`](#set_hooks) function. Once added, these hooks can be executed by anyone using the [`act`](#act) function. Mandatory hooks, marked with the duty flag, are executed during the fee collection process using the [`duty_act`](#duty_act) function.
+
+
+---
+
+
+## **Compensation Strategy**
+
+Each `hook` includes a `compensation_strategy` that defines how and when the executor of the hook will be compensated. This ensures that there is an incentive to call hooks according to predefined rules.
+
+The `CompensationStrategy` consists of the following values:
+
+```vyper
+struct CompensationStrategy:
+    amount: uint256  # In case of Dutch auction max amount
+    cooldown: CompensationCooldown
+    start: uint256
+    end: uint256
+    dutch: bool
+
+struct CompensationCooldown:
+    duty_counter: uint64  # last compensation epoch
+    used: uint64
+    limit: uint64  # Maximum number of compensations between duty acts (week)
+```
+
+- **`amount`**: The maximum amount of compensation available for executing the hook.
+- **`CompensationCooldown`**: Contains details about the cooldown period between compensations, consisting of:
+    - `duty_counter`: Keeps track of the last epoch in which a compensation was made.
+    - `used`: Indicates the number of compensations made within the current cooldown period.
+    - `limit`: Represents the maximum number of compensations allowed within the cooldown period.
+- **`start`**: Defines the starting time of the compensation period within a week.
+- **`end`**: Defines the ending time of the compensation period within a week.
+- **`dutch`**: A boolean indicating if the compensation uses a Dutch auction mechanism. If `true`, the compensation amount decreases linearly over time from the `start` to the `end`. This encourages earlier execution of the hook to receive a higher reward.
+
+
+---
+
+
+## **Hooks**
+
+Before hooks can be executed, they need to be added via `set_hooks`. These hooks can then be externally executed by anyone.
+
+
+### `hooks`
+!!! description "`Hooker.hooks(arg0: uint256) -> Hook: view`"
+
+    Getter for the hooks recorded in the contract.
+
+    Returns: `Hook` struct consisting of the target address (`address`), a byte array containing the method identifier and additional data (`Bytes[1024]`), compensation strategy (`CompensationStrategy`) consisting of "todo" and if the hook is a duty hook or not (`bool`).
```

Comment:
>"todo"

Not sure if you missed this or are still awating clarification, thought i'd flag on the off chance you forgot to come back to it.